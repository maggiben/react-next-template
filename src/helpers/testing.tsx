import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};