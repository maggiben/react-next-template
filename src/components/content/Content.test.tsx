import React from 'react';
import { renderer } from '@helpers/testing';
import 'jest-styled-components';
import Content from '.';

import { expect } from '@jest/globals';

test('Check Content works', () => {
    const tree = renderer.create(<Content />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('expect 1 to be 1', () => {
    expect(1).toBe(1);
});
