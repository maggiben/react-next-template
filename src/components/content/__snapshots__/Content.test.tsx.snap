// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Content works 1`] = `
.c2 {
  font-size: 16px;
  line-height: 24px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  font-family: Roboto;
}

.c1 {
  font-size: 32px;
  line-height: 42px;
  font-weight: 600;
  font-family: Work Sans;
}

.c0 {
  display: -webkit-box;
  display: -webkit-fl bex;
  display: -ms-fl bexbox;
  display: fl bex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100vh;
  gap: 8px;
  text-align: center;
}

.c0 img {
  margin: 0 auto 10px;
}

<div
  className="c0"
>
  <img
    alt="Búmeran logo"
    data-nimg="1"
    decoding="async"
    fetchpriority="high"
    onError={[Function]}
    onLoad={[Function]}
    src=""
    style={
      {
        "color": "transparent",
      }
    }
  />
  <div
    aria-level={2}
    className="c1"
    role="heading"
    size="xl"
  >
    React Búmeran Template
  </div>
  <span
    className="c2"
    size="l"
  >
    A Búmeran theme to start building a new app.
  </span>
</div>
`;
